src/main/java/com/Elessar/config/ClientConfig.java

```diff
@@ -0,0 +1,52 @@
+package com.Elessar.config;
+
+import com.Elessar.app.client.BlockingMsgQueue;
+import com.Elessar.app.client.KafkaMsgQueue;
+import com.Elessar.app.client.MsgQueue;
+import com.Elessar.app.util.MetricManager;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.*;
+
+@Configuration
+@ComponentScan("com.Elessar.app.client")
+@PropertySource("file:/Users/hans/Self-Learning/Project/mini-wechat/target/client.properties")
```

Comment:
Should not do this otherwise ev erybody other than you have to make a code change to build...

Try this:
@PropertySource("file:${client.root}/client.properties")
java  -Dclient.root=/Users/hans/Self-Learning/Project/mini-wechat/target -jar ./mini-wechat-client-console.jar

Same for server